bucket_map/src/index_entry.rs

```diff
@@ -58,7 +58,8 @@ impl IndexEntry {
     }
 
     pub fn data_bucket_from_num_slots(num_slots: Slot) -> u64 {
-        (num_slots as f64).log2().ceil() as u64 // use int log here?
+        // Compute the ceiling of log2 for integer
+        64 - ((num_slots - 1).leading_zeros()) as u64
```

Comment:
Are there any concerns if `num_slots` is 0? Is wrapping correct? (I see there isn't a test case if `num_slots` is 0.)

If wrapping is desired, `num_slots` can be a [`Wrapping<Slot>`](https://doc.rust-lang.org/std/num/struct.Wrapping.html) to indicate desired behavior. If `num_slots` cannot be zero, then [`NonZeroU64`](https://doc.rust-lang.org/std/num/struct.NonZeroU64.html) can be used.

Reply:
Good question. I think *num_slots* is slice size, should always be above 0? @jeffwashington 
If num_slots is zero, `(num_slots as f64).log2().ceil() as u64` will return 0. The new implementation will panic. 

If num_slots must be above zero, we could assert here
https://github.com/solana-labs/solana/blob/d8c45a69c3d0427898883ae03f320516d886f008/bucket_map/src/bucket.rs#L245

Reply:
A little bit later (after your new assert), there's this:
https://github.com/solana-labs/solana/blob/fe17865bd82fd927280e3ae7c02d5782bdd7ba65/bucket_map/src/bucket.rs#L293-L298

Which implies to me that `num_slots` _can_ be zero (which comes from `data.len()`). I don't remember the context off the top of my head what `num_slots == 0` is used for, but I think the current impl _does_ allow `data.len() == 0`.

Reply:
yup. I discovered that too.
Also there is a test that catches this too. 
`bucket_map::tests::bucket_map_test_update_to_0_len`.
I have pushed a commit to return 0 for empty slice, which keeps the behavior the same as before.